File: CPP_JAVA/Factorial.java
import java.util.Scanner;

class Factorial
{
    public static void main(String args[])
    {
        Scanner s = new Scanner(System.in);
        System.out.println("Enter the Number : ");
        int num = s.nextInt();
        int factorial = fact(num);
        System.out.println("Factorial of entered number is : " + factorial);
        s.close();
    }

    static int fact(int n)
    {
        int output;
        if(n == 1)
        {
            return 1;
        }
        else
        {
            output = n * fact(n-1);
            return output;
        }
    }
}

File: factorial.cpp
#include <iostream>

using namespace std;

double fact(double n)
{
    int fac;
    if (n == 0)
    {
        return 1;
    }
    else
    {
        fac = fact(n - 1) * n;
    }
    return fac;
}

int main ()
{
    cout << "Enter the number for factorial : ";
    double n;
    cin >> n;
    cout << endl;

    cout << "Factorial is : " << fact(n);
    return 0;
}